### PR TITLE
Task 7: My customer Api progress for the second parameter

### PR DESCRIPTION
**Pull request by kwarnerredhat**

### CODE REVIEW

File: src/api/CustomerApi.ts [hunk 1]
```diff
@@ -14,27 +14,27 @@ export interface Customer {
...
-export const addNewCustomer = (customer: Customer) => {
+export const addNewCustomer = (customer: Customer): Customers | undefined => {
   const customers = customersGetter();
   customers.push(customer);
   return customersSetter(customers);
 };
 
 // Super real API calls
-export const getCustomers = () => callAPI(customersGetter);
+export const getCustomers = () => callAPI(customersGetter, 0.7);
```
Comment (jlsherrill) [inline src/api/CustomerApi.ts:35]: this is the only place that we really need a 0.7, the other callApi()'s really don't need it

File: src/api/CustomerApi.ts [hunk 2]
```diff
@@ -14,27 +14,27 @@ export interface Customer {
...
-export const customersSetter = (customers: Customers | undefined) => {
+export const customersSetter = (customers: Customers | undefined): Customers | undefined => {
   localStorage.setItem('customers', JSON.stringify(customers));
   return customers;
 };
 
 export const customersGetter = (): Customers => {
   const customers = localStorage.getItem('customers');
-  if (customers) return [...JSON.parse(customers)];
+  if (customers) return JSON.parse(customers);
```
Comment (jlsherrill) [inline src/api/CustomerApi.ts]: curious why this change was needed?

File: src/api/CustomerApi.ts [hunk 3]
```diff
@@ -14,27 +14,27 @@ export interface Customer {
...
   return [];
 };
 
-export const addNewCustomer = (customer: Customer) => {
+export const addNewCustomer = (customer: Customer): Customers | undefined => {
```
Comment (jlsherrill) [inline src/api/CustomerApi.ts]: why was this change needed?

File: src/api/CustomerApi.ts [hunk 4]
```diff
@@ -14,27 +14,27 @@ export interface Customer {
 export type Customers = Array<Customer>;
 
 // Local storage "DAO layer" Getter/Setters
-export const customersSetter = (customers: Customers | undefined) => {
+export const customersSetter = (customers: Customers | undefined): Customers | undefined => {
```
Comment (jlsherrill) [inline src/api/CustomerApi.ts]: why was this change needed?

File: src/pages/Landing/index.tsx [hunk 5]
```diff
@@ -39,10 +39,17 @@ export default () => {
...
+  const { isLoading, data, error } = useQuery(
     'customers',
     getCustomers,
-    // TODO: Stretch - Use the options object to handle errors.
+    {
+      onError: (err) => {
+        console.error('Error occurred while fetching customers:', err);
+      },
+      retry: 3,
+      refetchOnWindowFocus: false,
```
Comment (jlsherrill) [inline src/pages/Landing/index.tsx]: i'm curious, what drew you to add ` refetchOnWindowFocus: false,` ?

Comment (kwarnerredhat) [reply src/pages/Landing/index.tsx]: i needed something to disable automatic refetching

Comment (jlsherrill) [reply src/pages/Landing/index.tsx]: why? was it causing a problem?

**Review by jlsherrill — APPROVED** [review]
